## Notes on how to follow the notebooks to replicate 16S rRNA data analysis

The notebooks in this directory contain all the instructions for 16S rRNA sequence cleaning in Unix, quality trimming and filtering with dada2, inference of amplicon sequence variants (AVS, inferred single 16S rRNA sequences), taxonomy inference with DECIPHER, diversity assessments and taxonomic summaries in R. For amplicon data analysis such as the one performed here, it is necessary to follow them in order.

The bioinformatic pipelines were modified from tutorials by Mike Lee (https://astrobiomike.github.io/amplicon/) and Daniel Utter (Orphan Lab at Caltech).

The raw data for the analyses can be found in the folder data/raw/16S_rRNA_seqs.

